pkg/cli/client.go

```diff
@@ -297,6 +300,25 @@ func (p *requestsPool) doJob(ctx context.Context, job *manifestJob) {
 		p.outputCh <- stringResult{"", err}
 	}
 
+	isSigned := false
+	cosignTag := strings.Replace(digestStr, ":", "-", 1) + "." + remote.SignatureTagSuffix
+
+	_, err = makeGETRequest(ctx, *job.config.servURL+"/v2/"+job.imageName+
+		"/manifests/"+cosignTag, job.username, job.password,
+		*job.config.verifyTLS, *job.config.debug, &job.manifestResp, job.config.resultWriter)
+	if err == nil {
+		isSigned = true
+	}
+
+	if !isSigned {
+		_, err = makeGETRequest(ctx, fmt.Sprintf("%s/oras/artifacts/v1/%s/manifests/%s/referrers",
+			*job.config.servURL, job.imageName, digestStr), job.username, job.password,
+			*job.config.verifyTLS, *job.config.debug, &job.manifestResp, job.config.resultWriter)
```

Comment:
I think we should check also the type of artefact is referring to this image. To know it's an actual signature.